lib/autokey/scripting/window.py

```diff
@@ -355,15 +364,10 @@ def get_window_geom(self, window_id):
         @return: C{[offsetx, offsety, sizex, sizey]} Returns none if no matches are found
         """
 
-        windowList = self._run_wmctrl(["-lG"])[1].split("\n")
-        output = []
-        for window in windowList:
-            w_split= window.split("  ")
-            hexid = w_split[0]
-            geom1 = w_split[1].split(" ")[1:]
-            geom2 = w_split[2].split(" ")
-
-
-            if hexid==window_id:
-                return [geom1+geom2]
-        return None
\ No newline at end of file
+        returncode, output = self._run_wmctrl(["-lG"])
+        regex = r"^(0x[0-9a-fA-F]{8})\s*(\d*)\s*(\d*)\s*(\d{1,})\s*(\d{1,})\s*(\d{1,})\s*(.*?)$"
+        matches = re.findall(regex, output, re.MULTILINE)
+        for match in matches:
+            if match[0] == window_id:
+                return [int(i) for i in match[2:-1]] # trim hexid, gravity on the front end and title on the backend
```

Comment:
What do you mean by gravity, front end, back end? Can you give an example in this comment? (also put it the line above, not inline)

Reply:
Gravity is a weird thing that is in the X11 window manager spec but appears to not be used/respected by at least the WM I use (whatever the Ubuntu 20.04 defaults to).

You can read more about it here https://specifications.freedesktop.org/wm-spec/wm-spec-1.3.html#Window%20Geometry

From reading about it, gravity seems like it would be an ideal way to center windows using `wmctrl`, my understanding is that it changes the "anchor" point where the window is placed, defaulting to the top left hand corner as one might expect, and each different value choosing a different point on either the midpoints or corners of the sides or the center itself

I may be reading it wrong, there are lots of words there lol. There seems to be some difference between the "client window" and the "frame window" but I'm not sure what it is referring to. 

I would include gravity in the return values for this function, as it does seem relevant to window geometry but the pre-existing `get_active_geometry` only returns the offset x,y and width/height. So for consistency I choose to discard it.

Not aware of anyone ever asking/wondering where it is, and from my googling of it it does not come up a lot.

Reply:
It should be considered that the two methods (`get_window_geom` and `get_active_geometry`) are more or less equivalent, I'd advocate that we should make `get_active_geometry` as deprecated and change up `get_window_geom` to accomplish the same thing (would get the active window by default, and take title/hex id as optional args)

The two functions are doing essentially the same thing (calling wmctrl and parsing the output), just `get_active` is limited to only getting the active window's geometry obviously.

Reply:
Rather than deprecate, how about you just convert it into a wrapper than calls your func?